rag_experiment_accelerator/config/index_config.py

```diff
@@ -1,59 +1,78 @@
-from dataclasses import dataclass
-from typing import Any
+from dataclasses import dataclass, field
 
-from rag_experiment_accelerator.embedding.embedding_model import EmbeddingModel
+from rag_experiment_accelerator.config.base_config import BaseConfig
+from rag_experiment_accelerator.config.chunking_config import ChunkingConfig
+from rag_experiment_accelerator.config.embedding_model_config import (
+    EmbeddingModelConfig,
+)
+from rag_experiment_accelerator.config.sampling_config import SamplingConfig
 
 
 @dataclass
-class IndexConfig:
+class IndexConfig(BaseConfig):
     """A class to hold parameters for each index configured through Config.
 
     Attributes:
         index_name_prefix (str):
             Prefix to use for the index created in Azure Search.
-        preprocess (bool):
-            Whether to preprocess the text before indexing.
-        chunk_size (int):
-            Chunk size for chunking documents.
-        overlap_size (int):
-            Overlap size for chunking documents.
-        embedding_model (int):
-            Embedding model to use for this config.
         ef_construction (int):
             Parameter ef_construction for HNSW index.
         ef_search (int):
             Parameter ef_search for HNSW index.
+        chunking (ChunkingConfig):
+            Configuration for chunking documents.
+        embedding_model (EmbeddingModelConfig):
+            Configuration for the embedding model.
+        sampling (SamplingConfig):
+            Configuration for sampling documents.
     """
 
-    index_name_prefix: str
-    preprocess: bool
-    chunk_size: int
-    overlap_size: int
-    embedding_model: EmbeddingModel
-    ef_construction: int
-    ef_search: int
-    sampling_percentage: int = 0
-    generate_title: bool = False
-    generate_summary: bool = False
-    override_content_with_summary: bool = False
+    index_name_prefix: str = "idx"
+    ef_construction: int = 400
+    ef_search: int = 400
+    chunking: ChunkingConfig = field(default_factory=ChunkingConfig)
+    embedding_model: EmbeddingModelConfig = field(default_factory=EmbeddingModelConfig)
+    sampling: SamplingConfig = field(default_factory=SamplingConfig)
+
+    def label_properties(self) -> dict:
+        """
+        Returns properties subset used for labeling.
+        """
+        properties = {
+            "idx": self.index_name_prefix,
```

Comment:
Same for these constants - "idx", "efc", "efs", "emn", "sp" - should be named constants

Reply:
Fixed